cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -101,26 +109,21 @@ def pad_index_two(index_string: str, reverse_complement: bool) -> str:
 
 def adapt_indexes(
     samples: List[FlowCellSample],
-    control_software_version: str,
-    reagent_kit_version: str,
-    expected_index_length: int,
+    run_parameters: RunParameters,
 ) -> None:
     """Adapts the indexes: pads all indexes so that all indexes have a length equal to the
     number  of index reads, and takes the reverse complement of index 2 in case of the new
     novaseq software control version (1.7) in combination with the new reagent kit
     (version 1.5)
     """
     LOG.info("Fix so that all indexes are on the correct format")
-    reverse_complement: bool = is_reverse_complement(
-        control_software_version=control_software_version,
-        reagent_kit_version_string=reagent_kit_version,
-    )
+    reverse_complement: bool = is_reverse_complement(run_parameters=run_parameters)
     for sample in samples:
         index1, index2 = sample.index.split("-")
         index1: str = index1.strip()
         index2: str = index2.strip()
         index_length = len(index1)
-        if expected_index_length == 10 and index_length == 8:
+        if run_parameters.index_length == 10 and index_length == 8:
```

Comment:
Might be hard for you to know, but what do these numbers (10 and 8) mean? Maybe we can extract them to some constants with more descriptive names

Reply:
These are the lengths in nucleotides of the indexes. I agree it is not very self-explanatory. Maybe improving this will be scope of another PR.